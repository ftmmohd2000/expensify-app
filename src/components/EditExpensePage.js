import React from 'react'
import ExpenseForm from './ExpenseForm'
import { connect } from 'react-redux'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  }
  onClick = () => {
    this.props.startRemoveExpense(this.props.expense.id)
    this.props.history.push('/')
  }
  render() {
    return (
      <div>
        <h1>Edit Expenses</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
          expense={this.props.expense}
        />
        <button onClick={this.onClick}>Remove</button>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(({ id }) => id === props.match.params.id)
})

const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (id) => dispatch(startRemoveExpense({ id }))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)
