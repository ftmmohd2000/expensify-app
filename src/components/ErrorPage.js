import React from 'react'
import { Link } from 'react-router-dom'

const ErrorPage = () => {
  return (
    <div>
      404 Page Not Found - <Link to="/">Back to HomePage</Link>
    </div>
  )
}

export default ErrorPage
