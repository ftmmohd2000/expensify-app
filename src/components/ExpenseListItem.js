import React from 'react'
import { removeExpense } from '../actions/expenses'
import { Link } from 'react-router-dom'

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  return (
    <div>
      <Link to={`/edit/${id}`}>{description}</Link>
      <p>{amount} - {createdAt}</p>
    </div>
  )
}

export default ExpenseListItem
