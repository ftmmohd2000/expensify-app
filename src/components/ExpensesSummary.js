import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import getExpensesTotal from '../selectors/expenses-total'
import expenseSelector from '../selectors/expenses'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
  <div>
    <p>You have {expenseCount} {`expense(s)`}</p>
    <p>With a total of {numeral(expensesTotal).format('$0,0.00')}</p>
  </div>
)

const mapStateToProps = (state) => {
  const filteredExpenses = expenseSelector(state)
  return {
    expenseCount: filteredExpenses.length,
    expensesTotal: getExpensesTotal(filteredExpenses) / 100
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
