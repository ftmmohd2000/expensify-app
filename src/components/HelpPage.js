import React from 'react'

function HelpPage() {
  return (
    <div>
      This is from fourth component
    </div>
  )
}

export default HelpPage
