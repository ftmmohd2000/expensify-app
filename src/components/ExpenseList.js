import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => {
  let a = 0
  return (
    <div>
      {
        props.expenses.length === 0 ? (
          <p>No Expenses</p>
        ) : (
            props.expenses.map((expense) => (
              <ExpenseListItem key={a++} {...expense} />
            ))
          )
      }
    </div>
  )
}

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state)
})

export default connect(mapStateToProps)(ExpenseList)

