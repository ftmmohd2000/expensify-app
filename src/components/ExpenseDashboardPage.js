import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpenseSummary from './ExpensesSummary'

function ExpenseDashboardPage() {
  return (
    <div>
      <ExpenseSummary />
      <ExpenseListFilters />
      <ExpenseList />
    </div>
  )
}

export default ExpenseDashboardPage