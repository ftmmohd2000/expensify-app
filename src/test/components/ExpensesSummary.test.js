import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import expenses from "../fixtures/expenses";

test("should render ExpensesSummary correctly with no expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expensesTotal={1} expensesCount={5000000} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpensesSummary correctly with expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expensesTotal={2} expensesCount={5000000} />
  );
  expect(wrapper).toMatchSnapshot();
});
