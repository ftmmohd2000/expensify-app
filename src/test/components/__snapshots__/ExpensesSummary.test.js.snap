// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with expenses 1`] = `
<div>
  <p>
    You have 
     
    expense(s)
  </p>
  <p>
    With a total of 
    $2.00
  </p>
</div>
`;

exports[`should render ExpensesSummary correctly with no expenses 1`] = `
<div>
  <p>
    You have 
     
    expense(s)
  </p>
  <p>
    With a total of 
    $1.00
  </p>
</div>
`;
